packages/ods/src/components/input/src/components/ods-input/ods-input.scss

```diff
@@ -0,0 +1,100 @@
+@use '~@ovhcloud/ods-themes/src/default/variables' as theme;
+@use '../../../../../style/focus';
+@use '../../../../../style/input';
+
+:host(.ods-input) {
+  position: relative;
+}
+
+$ods-input-button-right: 4px;
+$ods-input-button-top: -1px;
+
+.ods-input {
+  &__button {
+    $button: &;
+
+    display: inline-flex;
+    position: absolute;
+    top: $ods-input-button-top;
+    right: $ods-input-button-right;
+    border: none;
+    border-radius: 2px;
+    background: none;
+    cursor: pointer;
+    color: theme.$ods-color-primary-500;
+
+    &:focus-visible {
+      @include focus.ods-focus();
+    }
+
+    &:hover {
+      background-color: theme.$ods-color-primary-100;
+    }
+
+    &:active {
+      background-color: theme.$ods-color-primary-200;
+    }
+
+    &__clearable {
+      color: theme.$ods-color-neutral-600;
+
+      &:has(+ #{$button}) {
+        right: 30px;
+      }
+
+      &:hover {
+        background-color: theme.$ods-color-neutral-100;
+      }
+
+      &:active {
+        background-color: theme.$ods-color-neutral-200;
+      }
+    }
+  }
+
+  &__input {
+    $input: &;
+
+    @include input.ods-input();
+
+    &:not(#{$input}--error, #{$input}--disabled) {
+      &:focus {
+        border-color: theme.$ods-form-element-border-color-focused;
+      }
+
+      &:hover {
+        border-color: theme.$ods-form-element-border-color-hover;
+      }
+    }
+
+    &--disabled {
```

Comment:
can be moved into the mixin using native pseudo-selector `&:disabled` and then remove the class from the template
(when possible it's better to use native pseudo, this way people get it when using the mixin without depending on BEM class name)